tests/test_audio.py

```diff
@@ -48,6 +52,35 @@ def test_get_integrated_lufs():
         assert i == li
 
 
+def test_match_sample_length():
+    durations_to_match = [1, 2, 5, 7, 22500, 44100, 88200, 100001]
+    tmpfiles = []
+    with _close_temp_files(tmpfiles):
+        carhorn = tempfile.NamedTemporaryFile(suffix='.wav', delete=True)
+        shutil.copyfile(CARHORN_FILE, carhorn.name)
+        tmpfiles.append(carhorn)
+
+        siren = tempfile.NamedTemporaryFile(suffix='.wav', delete=True)
+        shutil.copyfile(SIREN_FILE, siren.name)
+        tmpfiles.append(siren)
+
+        for _duration in durations_to_match:
+            match_sample_length(carhorn.name, _duration)
+            carhorn_audio, _ = sf.read(carhorn.name)
+            assert carhorn_audio.shape[0] == _duration
+
+            match_sample_length(siren.name, _duration)
+            siren_audio, _ = sf.read(siren.name)
+            assert siren_audio.shape[0] == _duration
+
+            # should be summable
+            summed_events = sum([carhorn_audio, siren_audio])
+            assert summed_events.shape[0] == _duration
+        
+        pytest.raises(ScaperError, match_sample_length, carhorn.name, -1)
+        pytest.raises(ScaperError, match_sample_length, carhorn.name, 0)
```

Comment:
It might be more systematic to place the invalid values in a list and iterate on it (as with the valid durations above). In addition to 0 and -1, we should check for floats (invalid) too, as it's possible for people to make the mistake of inserting a duration value in seconds rather than samples and we should catch this.